backend/src/ee/services/dynamic-secret-lease/dynamic-secret-lease-service.ts

```diff
@@ -259,7 +274,25 @@ export const dynamicSecretLeaseServiceFactory = ({
     if (!dynamicSecretLease || dynamicSecretLease.dynamicSecret.folderId !== folder.id)
       throw new NotFoundError({ message: `Dynamic secret lease with ID '${leaseId}' not found` });
 
-    const dynamicSecretCfg = dynamicSecretLease.dynamicSecret;
+    const dynamicSecretCfg = await dynamicSecretDAL.findOneWithMetadata({
```

Comment:
@sheensantoscapadngan I think you override the `fineOne` rather than adding another function. Because as metadata is part of permission now, it will be needed through to check permission. I guess we can support it in by default